program-runtime/src/loaded_programs.rs

```diff
@@ -425,6 +425,15 @@ impl LoadedPrograms {
 
                             if current_slot >= entry.effective_slot {
                                 return Some((key, entry.clone()));
+                            } else {
```

Comment:
Wouldn't this condition also trigger when recompilation for feature set changes happens?

Reply:
That's correct. One potential solution is to define a const for the gap between deployed and effective slots for the delay visible programs. e.g. 1 slot. We can return the tombstone only if the gap is 1 slot, and the current slot >= deployment slot but < effective slot.

Any other solutions to handle this?

Reply:
Updated the code and the test.